src/main/scala/mesosphere/marathon/core/appinfo/GroupInfoService.scala

```diff
@@ -0,0 +1,17 @@
+package mesosphere.marathon.core.appinfo
+
+import mesosphere.marathon.state.Group
+
+import scala.concurrent.Future
+
+/**
+  * Queries for extended group information.
+  */
+trait GroupInfoService {
+
+  /**
+    * Query info for an existing group.
+    */
+  def queryForGroup(group: Group, appEmbed: Set[AppInfo.Embed], groupEmbed: Set[GroupInfo.Embed]): Future[GroupInfo]
```

Comment:
Generally I agree with your point. The way we handle this in the AppsResource is to implement a special AppSelector. We could implement something like a GroupSelector which extends AppSelector.
But I see this in a separate PR.